lib/src/main/java/com/glung/redux/Middlewares.java

```diff
@@ -0,0 +1,75 @@
+package com.glung.redux;
+
+import java.util.Arrays;
+import java.util.List;
+
+import redux.api.*;
+import redux.api.Store;
+import redux.api.enhancer.Middleware;
+
+public class Middlewares {
+
+    private static class MiddlewareStore<S> implements redux.api.Store<S> {
+
+        private final redux.api.Store<S> nextStore;
+        private final List<redux.api.enhancer.Middleware<S>> middlewares;
+
+        public MiddlewareStore(redux.api.Store<S> nextStore,
+                List<redux.api.enhancer.Middleware<S>> middlewares) {
+            this.nextStore = nextStore;
+            this.middlewares = middlewares;
+        }
+
+        @Override
+        public S getState() {
+            return nextStore.getState();
+        }
+
+        @Override
+        public Subscription subscribe(Subscriber subscriber) {
+            return nextStore.subscribe(subscriber);
+        }
+
+        @Override
+        public void replaceReducer(Reducer<S> reducer) {
+            nextStore.replaceReducer(reducer);
+        }
+
+        @Override
+        public Object dispatch(Object action) {
+            return createMiddlewareDispatcher(middlewares).dispatch(action);
```

Comment:
Do you think we could create the Dispatcher it only once by calling `createMiddlewareDispatcher` in `applyMiddlewares` ?